### PR TITLE
Implemented fixes in water properties computation (density, enthalpy)

### PR DESCRIPTION
**Pull request by francoishamon**
This PR implements small fixes that we found while validating water properties for SPE11:
- [X] Corrected wrong value in the table of saturated water density as a function of temperature (mostly likely an old copy-paste that went wrong for T = 40 degC)
- [X] Added missing reference water-phase enthalpy at T_0 = 273.15 K 
- [X] Added an warning message if someone tries to use the Phillips model outside of its range of validity (low salinity)
- [X] Implemented a bypass to make sure that people using the Phillips model with a zero salinity (frequent case) get valid results

Rebaseline done in https://github.com/GEOS-DEV/integratedTests/pull/46

### CODE REVIEW

**Comment by paveltomin** [comment]
can you share a density comparison for zero salinity case?
just wondering how big is the difference

**Comment by francoishamon** [comment]
> can you share a density comparison for zero salinity case? just wondering how big is the difference

Before the fix, we had something quite inaccurate because the Phillips model is not meant to be used in these conditions (Ezrokhi is):

Screenshot 2023-09-18 at 3 54 33 PM

After the fix, we are in good agreement with NIST:

Screenshot 2023-09-18 at 3 54 44 PM

Note that I made a typo in the legend of the two images and pressure is in Pa.

To get an even better match and fully match NIST, the path is just to replace the hard-coded water compressibility (see [here](https://github.com/GEOS-DEV/GEOS/blob/b897571b0ef418527fbe8ecf11448f4583e372b7/src/coreComponents/constitutive/fluid/multifluid/CO2Brine/functions/EzrokhiBrineDensity.cpp#L60) for Ezrokhi and [here](https://github.com/GEOS-DEV/GEOS/blob/b897571b0ef418527fbe8ecf11448f4583e372b7/src/coreComponents/constitutive/fluid/multifluid/CO2Brine/functions/PhillipsBrineDensity.cpp#L105) for Phillips) with a [table](https://www.researchgate.net/publication/234849778_Compressibility_of_Water_as_a_Function_of_Temperature_and_Pressure/link/55159b9b0cf2b5d6a0ea6347/download?_tp=eyJjb250ZXh0Ijp7InBhZ2UiOiJwdWJsaWNhdGlvbiIsInByZXZpb3VzUGFnZSI6bnVsbH19) of compressibility as a function of pressure and temperature. It would be good to do it because it has such a big impact on the compressibility of the system, and it is easy to do (in a separate PR).

With salinity, we are in good shape

**Comment by jafranc** [comment]
> > can you share a density comparison for zero salinity case? just wondering how big is the difference
> 
> Before the fix, we had something quite inaccurate because the Phillips model is not meant to be used in these conditions (Ezrokhi is):
> 
> Screenshot 2023-09-18 at 3 54 33 PM
> After the fix, we are in good agreement with NIST:
> 
> Screenshot 2023-09-18 at 3 54 44 PM
> Note that I made a typo in the legend of the two images and pressure is in Pa.
> 
> To get an even better match and fully match NIST, the path is just to replace the hard-coded water compressibility (see [here](https://github.com/GEOS-DEV/GEOS/blob/b897571b0ef418527fbe8ecf11448f4583e372b7/src/coreComponents/constitutive/fluid/multifluid/CO2Brine/functions/EzrokhiBrineDensity.cpp#L60) for Ezrokhi and [here](https://github.com/GEOS-DEV/GEOS/blob/b897571b0ef418527fbe8ecf11448f4583e372b7/src/coreComponents/constitutive/fluid/multifluid/CO2Brine/functions/PhillipsBrineDensity.cpp#L105) for Phillips) with a [table](https://www.researchgate.net/publication/234849778_Compressibility_of_Water_as_a_Function_of_Temperature_and_Pressure/link/55159b9b0cf2b5d6a0ea6347/download?_tp=eyJjb250ZXh0Ijp7InBhZ2UiOiJwdWJsaWNhdGlvbiIsInByZXZpb3VzUGFnZSI6bnVsbH19) of compressibility as a function of pressure and temperature. It would be good to do it because it has such a big impact on the compressibility of the system, and it is easy to do (in a separate PR).
> 
> With salinity, we are in good shape

For later reference as the research-gate links seems broken.
[Compressibility of water as a function of temperature and pressure; Rana A. Fine, Frank J. Millero](https://doi.org/10.1063/1.1679903)(tho not free access)

**Review by jafranc — APPROVED** [review]
Looks good critical fix ! Thank you @francoishamon

File: src/coreComponents/constitutive/fluid/multifluid/CO2Brine/functions/CO2Enthalpy.cpp [hunk 1]
```diff
@@ -269,12 +269,16 @@ CO2Enthalpy::calculateCO2Enthalpy( PTTableCoordinates const & tableCoords,
   localIndex const nPressures = tableCoords.nPressures();
   localIndex const nTemperatures = tableCoords.nTemperatures();
 
+  // Note that the enthalpy values given in Span and Wagner assume a reference enthalphy defined as: h_0 = 0 J/kg at T_0 = 298.15 K
```
[the remark this answers is not in the document]
Comment (dkachuma) [reply src/coreComponents/constitutive/fluid/multifluid/CO2Brine/functions/CO2Enthalpy.cpp:272]: I can't determine why. I can only assume that it was an ideal gas assumption of some sort necessary to put all the measurements from different sources on the same scale.

File: src/coreComponents/constitutive/fluid/multifluid/CO2Brine/functions/PhillipsBrineDensity.cpp [hunk 2]
```diff
@@ -93,7 +131,21 @@ TableFunction const * makeDensityTable( string_array const & inputParams,
...
+    GEOS_LOG_RANK_0_IF( salinity < 0.25,
+                        GEOS_FMT( "{}: Warning! The salinity value of {} is below the range of validity of the Phillips model, results may be inaccurate",
+                                  functionName, salinity ) );
+  }
+  else
+  {
+    // the Phillips correlation is inaccurate in the absence of salinity.
+    // since this is a very frequent case, we implement an alternate (more accurate) method below
+    calculatePureWaterDensity( tableCoords, functionName, densities );
+  }
```
Comment (dkachuma) [inline src/coreComponents/constitutive/fluid/multifluid/CO2Brine/functions/PhillipsBrineDensity.cpp:148]: I think I would still issue a message here that we are switching to pure water properties. There might be quite a jump in the density between zero salinity and a very small salinity value which some users may find unexplained.

**Review by dkachuma — APPROVED** [review]

**Pull request merged**